python-package/xgboost/spark/data.py

```diff
@@ -1,194 +1,85 @@
-# type: ignore
-"""Xgboost pyspark integration submodule for data related functions."""
-# pylint: disable=too-many-arguments
-from typing import Iterator
+from collections import defaultdict, namedtuple
+from typing import Any, Dict, Iterator, List, Optional, Sequence, Tuple
 
 import numpy as np
 import pandas as pd
 
+from xgboost.compat import concat
 from xgboost import DMatrix
 
 
-def _prepare_train_val_data(
-    data_iterator, has_weight, has_validation, has_fit_base_margin
-):
-    def gen_data_pdf():
-        for pdf in data_iterator:
-            yield pdf
+def stack_series(df: pd.Series) -> np.ndarray:
+    array = df.to_numpy(copy=False)
+    if array.ndim == 1:
+        array = array.reshape(array.shape[0], 1)
+    array = np.stack(array[:, 0])
+    return array
```

Comment:
I made a simple test:
```
>>> p1 = pd.DataFrame({'a':[[3,4],[5,6],[7,8]]})
>>> p1.a.to_numpy().shape
(3,)
>>> p1.a.to_numpy().reshape(3,1)
array([[list([3, 4])],
       [list([5, 6])],
       [list([7, 8])]], dtype=object)
```
After reshape, each element becomes list, does this hurt performance ?